### PR TITLE
Dark Mode: Implement Metrics

### PR DESCRIPTION
**Pull request by darkwing**
Initial implementation of metrics for themeing.

### CODE REVIEW

File: shared/constants/metametrics.js [hunk 1]
```diff
@@ -190,6 +190,7 @@ export const TRAITS = {
   NUMBER_OF_NFT_COLLECTIONS: 'number_of_nft_collections',
   OPENSEA_API_ENABLED: 'opensea_api_enabled',
   THREE_BOX_ENABLED: 'three_box_enabled',
+  THEME: 'theme',
```
Comment (brad-decker) [inline shared/constants/metametrics.js:193]: [req] Please add the entries to the two jsdoc types related to traits for this entry.

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 2]
```diff
@@ -261,7 +261,17 @@ export default class ExperimentalTab extends PureComponent {
               id="select-theme"
               options={themesOptions}
               selectedOption={theme}
-              onChange={async (newTheme) => setTheme(newTheme)}
+              onChange={(newTheme) => {
+                this.context.trackEvent({
+                  category: 'Settings',
+                  event: 'Theme Change',
```
Comment (brad-decker) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:267]: [req] similar to "Token Added" and "Clicked ...${x}" lets make this at least "Changed"

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 3]
```diff
@@ -261,7 +261,17 @@ export default class ExperimentalTab extends PureComponent {
...
+                  properties: {
+                    theme_selected: newTheme,
+                    legacy_event: true,
```
Comment (brad-decker) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:270]: [REQ] remove legacy_event, this is reserved for events that were a part of our old schema.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Just some minor requests, looks good otherwise.

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 4]
```diff
@@ -261,7 +261,16 @@ export default class ExperimentalTab extends PureComponent {
...
+              onChange={(newTheme) => {
+                this.context.trackEvent({
+                  category: 'Settings',
+                  event: 'Theme Changed',
+                  properties: {
+                    theme_selected: newTheme,
+                  },
+                });
+                setTheme(newTheme);
+              }}
```
Comment (jpuri) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:273]: I suspect an inline function here can cause re-rendering in `<Dropdown` `useCallback` may be a good idea here.

**Review by brad-decker — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**